File: Problems/206. Reverse Linked List.java

class ListNode {
    int val;
    ListNode next;
    ListNode() {}
    ListNode(int val) { this.val = val; }
    ListNode(int val, ListNode next) { this.val = val; this.next = next; }
}


class Solution {


    public ListNode reverseList(ListNode head) {

        ListNode cur = head;
        ListNode prev = null;
        ListNode next;

        while(cur != null){
            next = cur.next;
            cur.next = prev;
            prev = cur;
            cur = next;
        }

        head = prev;
        return head;
    }


}

File: 206. Reverse Linked List.py
class Solution:
    def reverseList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        
        p1 = None
        cur = head

        while cur != None:
            temp = cur.next
            cur.next = p1
            p1 = cur
            cur = temp

        return p1        

